src/stan/variational/advi.hpp

```diff
@@ -285,26 +414,48 @@ namespace stan {
 
             if (print_stream_)
               *print_stream_ << std::endl;
+
+            if (do_more_iterations == false &&
+                rel_difference_(elbo, elbo_best) > 0.05) {
+              if (print_stream_)
+                *print_stream_
+                  << "Informational Message: The ELBO at a previous "
+                  << "iteration is larger than the ELBO upon "
+                  << "convergence!"
+                  << std::endl
+                  << "This means that the variational approximation has "
+                  << "not converged to the global optima."
+                  << std::endl;
+            }
           }
 
-          // Check for max iterations
-          if (iter_counter == max_iterations) {
+          if (iter_main == max_iterations) {
             if (print_stream_)
-              *print_stream_ << "MAX ITERATIONS REACHED" << std::endl;
+              *print_stream_
+                << "Informational Message: The maximum number of "
+                << "iterations is reached! The algorithm has not "
+                << "converged."
+                << std::endl
+                << "Values from this variational approximation are not "
+                << "guaranteed to be meaningful."
+                << std::endl;
             do_more_iterations = false;
           }
 
-          ++iter_counter;
+          ++iter_main;
         }
       }
 
       /**
        * Runs the algorithm and writes to output.
        *
+       * @param  eta            eta parameter for stepsize scaling
        * @param  tol_rel_obj    relative tolerance parameter for convergence
        * @param  max_iterations max number of iterations to run algorithm
+       * @param  tuning_iter    number of iterations for hyperparameter tuning
        */
-      int run(double tol_rel_obj, int max_iterations) const {
+      int run(std::string eta, double tol_rel_obj,
```

Comment:
This should be `const std::string& eta` --- if you don't
make it a reference, it allocates more memory and copies on
every call.

> On Aug 26, 2015, at 11:22 AM, Alp Kucukelbir <notifications@github.com> wrote:
> 
> In src/stan/variational/advi.hpp:
> 
> >         */
> > -      int run(double tol_rel_obj, int max_iterations) const {
> > +      int run(std::string eta, double tol_rel_obj,
> 
> i added const.
> 
> testing for run... yeah. we can talk.
> 
> —
> Reply to this email directly or view it on GitHub.
>